services/osparc-gateway-server/Dockerfile

```diff
@@ -0,0 +1,170 @@
+
+# for ARM architecture this helps a lot VS building packages
+ENV PIP_EXTRA_INDEX_URL=https://www.piwheels.org/simple
+
+
+EXPOSE 8000
+
+
+# -------------------------- Build stage -------------------
+# Installs build/package management tools and third party dependencies
+#
+# + /build             WORKDIR
+#
+FROM base as build
+
+ENV SC_BUILD_TARGET=build
+
+RUN rm -f /etc/apt/apt.conf.d/docker-clean
+RUN  --mount=type=cache,id=basecache,target=/var/cache/apt,mode=0755,sharing=locked \
+  --mount=type=cache,id=baseapt,target=/var/lib/apt,mode=0755,sharing=locked \
+  set -eux \
+  && apt-get update \
+  && apt-get install -y --no-install-recommends \
+  build-essential \
+  git \
+  golang-go \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/*
+
+
+# NOTE: python virtualenv is used here such that installed packages may be moved to production image easily by copying the venv
+RUN python -m venv "${VIRTUAL_ENV}"
+RUN --mount=type=cache,mode=0777,target=/root/.cache/pip \
+  pip install --upgrade \
+  pip~=23.0  \
+  wheel \
+  setuptools
+
+WORKDIR /build
+
+# install base 3rd party dependencies (NOTE: this speeds up devel mode)
+COPY --chown=scu:scu services/osparc-gateway-server/requirements/_base.txt .
+RUN --mount=type=cache,mode=0777,target=/root/.cache/pip \
+  pip install \
+  --requirement _base.txt
+
+# in ARM64 mode there is a catch: the piwheels package does not contain the dask-gateway-proxy executable in 64-bit
+RUN dpkgArch="$(dpkg --print-architecture)";\
+  case "$dpkgArch" in \
+  arm64) git clone --depth 1 --branch 0.9.0 https://github.com/dask/dask-gateway.git \
+  && cd dask-gateway/osparc-gateway-server \
+  && pip install .\
+  ;; \
+  esac;
+
+# --------------------------Prod-depends-only stage -------------------
+# This stage is for production only dependencies that get partially wiped out afterwards (final docker image concerns)
+#
+#  + /build
+#    + services/osparc-gateway-server [scu:scu] WORKDIR
+#
+FROM build as prod-only-deps
+
+ENV SC_BUILD_TARGET=prod-only-deps
+
+COPY --chown=scu:scu services/osparc-gateway-server/ /build/services/osparc-gateway-server
+
+WORKDIR /build/services/osparc-gateway-server
+
+RUN --mount=type=cache,mode=0777,target=/root/.cache/pip \
+  pip install \
+  --requirement requirements/prod.txt
+
+# --------------------------Production stage -------------------
+# Final cleanup up to reduce image size and startup setup
+# Runs as scu (non-root user)
+#
+#  + /home/scu     $HOME = WORKDIR
+#    + services/osparc-gateway-server [scu:scu]
+#
+FROM base as production
+
+ENV SC_BUILD_TARGET=production \
+  SC_BOOT_MODE=production
+
+ENV PYTHONOPTIMIZE=TRUE
+
+WORKDIR /home/scu
+
+# bring installed package without build tools
+COPY --from=prod-only-deps --chown=scu:scu ${VIRTUAL_ENV} ${VIRTUAL_ENV}
+# copy docker entrypoint and boot scripts
+COPY --chown=scu:scu services/osparc-gateway-server/docker services/osparc-gateway-server/docker
+
+
+# TODO: Create healthcheck
```

Comment:
Commented code + TODO ?!

Reply:
yes this is bad. but was already there before.